### PR TITLE
LPS-146480 Include supported editors in sample portlet

### PR DESCRIPTION
**Pull request by john-co**

### CODE REVIEW

**Comment by john-co** [comment]
Hi @diegonvs, there's some unformatted content in `CKEditor BBCode Tag` and no content in `Editor React Component`. Is there something I'd need to do to update the tag to use them correctly? Would you be able to help fix this?

https://user-images.githubusercontent.com/35235266/167516411-42390e26-ce0c-469e-8d2c-9b20f3437825.mp4

This is the stacktrace I got after adding the sample portlet to the content page.
https://gist.github.com/john-co/0b84cde265666abe47d246e35da51d82

**Comment by john-co** [comment]
ci:test:frontend:nocompile

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/js/Editor.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {Editor} from 'frontend-editor-ckeditor-web';
```
Comment (markocikos) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/js/Editor.js:15]: We [deprecated](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/Editor.js#L25-L27) `Editor`, we should not add it to the sample module.

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/alloyeditor.jsp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	<p>
+		Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Nunc id cursus metus aliquam eleifend mi in nulla. Quam adipiscing vitae proin sagittis nisl rhoncus. Suspendisse faucibus interdum posuere lorem. Nullam ac tortor vitae purus faucibus ornare. Ac felis donec et odio pellentesque diam. Nulla at volutpat diam ut. Posuere urna nec tincidunt praesent semper feugiat nibh. Gravida quis blandit turpis cursus. Proin libero nunc consequat interdum varius. Sollicitudin ac orci phasellus egestas tellus rutrum tellus pellentesque. Neque volutpat ac tincidunt vitae semper quis lectus nulla at. Odio euismod lacinia at quis risus sed vulputate odio ut. Augue lacus viverra vitae congue eu consequat ac. Elementum sagittis vitae et leo duis ut diam. Diam quis enim lobortis scelerisque fermentum dui faucibus.
+	</p>
+
+	<p>
+		This paragraph contains a <a href="https://example.com">link</a>.
+	</p>
+
+	<img src="https://images.unsplash.com/photo-1539037116277-4db20889f2d4?fit=crop&w=300" />
+</liferay-util:buffer>
```
Comment (markocikos) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/alloyeditor.jsp:35]: I like the fact that all editors have the sample content. We can extract this `<liferay-util:buffer>` into `view.jsp`, and use it in partials, just adding titles on top.

**Review by markocikos — COMMENTED** [review]
@diegonvs see comments inline. Also, the commit still has `LPS-XXXXX` in message :slightly_smiling_face:

**Comment by diegonvs** [comment]
>  Is there something I'd need to do to update the tag to use them correctly? Would you be able to help fix this?

I don't know 😢 

I set the possible values for `editorName` and created a tab.

I don't remember if it worked before(when I created this sample), but we should investigate this behavior further. Can you fill a ticket, pls?

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	<p>
+		This paragraph contains a <a href="https://example.com">link</a>.
+	</p>
+
+	<img src="https://images.unsplash.com/photo-1539037116277-4db20889f2d4?fit=crop&w=300" />
+</liferay-util:buffer>
+
+<liferay-ui:input-editor
+	contents="<%= sampleEditorContents %>"
+	editorName="inputeditor"
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp:37]: Interesting, I didn't know the existence of these possible values 🤔 🤔🤔🤔🤔🤔🤔🤔🤔

**Comment by diegonvs** [comment]
@john-co will work on this when it's possible, and I'll guide him for things he doesn't know :)

**Comment by john-co** [comment]
Thanks @markocikos ! I'll take a closer look to see if I can make those suggested changes.

cc/ @john-co testing

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp (hunk 3 above)
[the remark this answers is not in the document]
Comment (john-co) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp:37]: removed file

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/js/Editor.js (hunk 1 above)
Comment (john-co) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/js/Editor.js:15]: removed file

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/alloyeditor.jsp (hunk 2 above)
Comment (john-co) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/alloyeditor.jsp]: Thanks, this part I'm not sure if I've done correctly with my updates. I'll follow up with Diego tomorrow on this.

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp (hunk 3 above)
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: Hey @markocikos, I'm starting to think those values are outdated. I remember we dropped [TinyMCE a few years ago](https://github.com/liferay/liferay-portal/blob/7.3.x/readme/BREAKING_CHANGES.markdown#tinymce-editor-is-no-longer-bundled-by-default) and when testing ckeditor_creole things don't work. Maybe we may configure things not just pass ckeditor_creole to editorName? 🤔

[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: https://user-images.githubusercontent.com/7663513/168392375-92da849f-a57a-4131-9901-d6275c104b19.mp4

[InlineEditor React component](https://cs.github.com/liferay/liferay-portal/blob/79e90f042922c7fa37ad3e185dcb5ac4d34f7ccf/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/index.js?q=InlineEditor#L18) seems not working too, without any errors on console 🤔 

This Editor is not used when performing [a code search](https://cs.github.com/liferay/liferay-portal?q=InlineEditor). Is this an opportunity to remove?

**Comment by diegonvs** [comment]
Hey @john-co, I fixed the "shared" sampleEditorContents via requests to partials, removed stale files, and added the ClassicEditor React component that would be good for testing.

See aac24acbb8a0e50b3daa932403d6ba6db869cd0e

**Comment by john-co** [comment]
ci:test:relevant

**Comment by john-co** [comment]
ci:test:sf

File: modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp (hunk 3 above)
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: Yeah, we should, at the very least, deprecate `InlineEditor`. It's an early POC of having a thin wrapper around CKEditor, along with  `Editor`. They were not good enough for all DXP needs. `Editor` was replaced by `ClassicEditor`, and `InlineEditor` by `Balloon Editor`.

This is unrelated to this PR, and to prior conversation in this comment thread. `inputeditor` is not `InlineEditor`. We do have `InputEditorTag`, but that's unrelated to CKEditor, that's a version of localized input tag.

[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: > I'm starting to think those values are outdated. I remember we dropped [TinyMCE a few years ago](https://github.com/liferay/liferay-portal/blob/7.3.x/readme/BREAKING_CHANGES.markdown#tinymce-editor-is-no-longer-bundled-by-default) and when testing ckeditor_creole things don't work. Maybe we may configure things not just pass ckeditor_creole to editorName?

Do you have a better, updated source of supported CKEditor versions? If not, we should make one!

[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: > Do you have a better, updated source of supported CKEditor versions? 

I don't understand this, sorry. Could you explain a bit more?

[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/inputeditor.jsp]: If you mean editors supported, I think only Dani and Pablo should have the complete context

**Comment by john-co** [comment]
Also, noticed that the CKEditor BB Code doesn't have the style formatting applied correctly. ie: still displays `<h1>` `<p>` tags instead of applying the styling.
Selection_250

**Comment by john-co** [comment]
cc/ @diegonvs , let me know if still able to help with this when you get the chance.

**Comment by diegonvs** [comment]
Sorry for not reaching this thread quickly. We talked about it somewhere, and I asked you to create a ticket. At least for this issue:

> noticed that the balloon editor doesn't display until I change tabs within the sample portlet

**Comment by john-co** [comment]
thanks, created a ticket https://issues.liferay.com/browse/LPS-154231

**Comment by john-co** [comment]
Hi @diegonvs, I'll close this PR for now since it's blocked by https://issues.liferay.com/browse/LPS-154231 and not sure when it'll be worked on.

**Comment by markocikos** [comment]
> Hi @diegonvs

:cry:

**Pull request closed without merging**